### PR TITLE
Add linear from unifold for a clean-path residual transformer

### PR DESCRIPTION
**Pull request by hypnopump**
Transformer layers were not following[ clean-path residual principles](https://arxiv.org/abs/1603.05027) (model starts at identity). This PR borrows the Linear module from UniFold and ensures all transformer residual layers start at 0 in training.

**Bonus**: modifies the random seed to affect all GPUs at the same time, preventing lack of reproducibility in multi-gpu setup

New models train faster now (blue is new, red is old): 
Captura de Pantalla 2023-12-12 a las 19 18 47

### CODE REVIEW

**Comment by hypnopump** [comment]
Previous param `update_freq` was doing the job of gradient accumulation but not dividing by the number of accumulated steps. Change it so it behaves like `gradient_accumulation` and *effective_batch_size* can be handled with the same learning rate depending on the number of gpus and memory consumption. 

To be fair, this could be a separate PR. Could implement it like this if it's wanted. 

See expected behaviour from [huggingface's accelerate](https://huggingface.co/docs/accelerate/usage_guides/gradient_accumulation): 
Captura de Pantalla 2023-12-16 a las 16 09 43
